test/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManagerSpec.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.actors
+
+import akka.actor.{ActorSystem, Props}
+import akka.testkit.{ImplicitSender, TestKit}
+import cats.data.EitherT
+import com.miguno.akka.testing.VirtualTime
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+import play.api.test.Helpers._
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.{AttemptTaxEnrolmentAllocationToGroup, RetryTaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.connectors.TaxEnrolmentConnector
+import uk.gov.hmrc.cgtpropertydisposals.models.Address.UkAddress
+import uk.gov.hmrc.cgtpropertydisposals.models.TaxEnrolmentRequest.{TaxEnrolmentFailed, TaxEnrolmentInProgress}
+import uk.gov.hmrc.cgtpropertydisposals.models.{Error, TaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.repositories.{DefaultTaxEnrolmentRetryRepository, MongoSupport, TaxEnrolmentRetryRepository}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class TaxEnrolmentRetryManagerSpec
+    extends TestKit(ActorSystem())
+    with ImplicitSender
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with MongoSupport
+    with MockFactory {
+
+  override def beforeEach: Unit =
+    mongo().drop()
+
+  override def afterAll: Unit =
+    TestKit.shutdownActorSystem(system)
+
+  val connector                  = mock[TaxEnrolmentConnector]
+  val mockRepo                   = mock[TaxEnrolmentRetryRepository]
+  val repository                 = new DefaultTaxEnrolmentRetryRepository(reactiveMongoComponent)
```

Comment:
do you need an actual instance of the repo?

Reply:
Yes as discussed yesterday. I will work on fixing this test this morning.